x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtAuthenticator.java

```diff
@@ -64,6 +67,14 @@ public JwtAuthenticator(
     }
 
     public void authenticate(JwtAuthenticationToken jwtAuthenticationToken, ActionListener<JWTClaimsSet> listener) {
+        SpecialPermission.check();
+        AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
+            doAuthenticate(jwtAuthenticationToken, listener);
+            return null;
+        });
```

Comment:
Why do we need `doPrivileged` here when its caller (`JwtRealm.authenticate`) already has a doPrivileged block? Is it for tests?

Also on the topic of needing to add `doPrivileged` again in listener: When signature validation fails, the authenticator can [trigger a reload](https://github.com/elastic/elasticsearch/blob/353f357f0b82cd7911b50d98e1da89401f931c46/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtSignatureValidator.java#L274) from HTTP URL. The subsequent validation will run in a different thread. Does it not require another doPrivileged block? But if all tests are passing, maybe it is not an issue.

Reply:
> Why do we need doPrivileged here when its caller (JwtRealm.authenticate) already has a doPrivileged block?

Good q! I forgot to add a note here. This `doPrivileged` is redundant and I went back and forth about including it vs. not. I have it in, largely to make the class more self-contained and be a little more robust to future refactors. Happy to leave it out though. Let me know which you prefer.

> When signature validation fails, the authenticator can [trigger a reload](https://github.com/elastic/elasticsearch/blob/353f357f0b82cd7911b50d98e1da89401f931c46/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtSignatureValidator.java#L274) from HTTP URL. The subsequent validation will run in a different thread. Does it not require another doPrivileged block?

This is a good catch, I think we do need a doPrivileged block there. I'll check this out 👀 The reason tests are passing is that unit tests don't properly exercise permission checks since JUnit has accessDeclaredMembers: https://github.com/elastic/elasticsearch/blob/main/server/src/main/resources/org/elasticsearch/bootstrap/test-framework.policy#L69-L74 

REST tests probably just don't exercise this path.